main/src/io/github/iltotore/iron/package.scala

```diff
@@ -77,3 +77,13 @@ extension [A](value: A)
    */
   inline def refineOption[B](using inline constraint: Constraint[A, B]): Option[A :| B] =
     Option.when(constraint.test(value))(value)
+
+  /**
+   * Refine the given value at runtime, assuming the constraint holds.
+   *
+   * @param constraint the constraint to test with the value to refine.
+   * @return a constrained value, without performing constraint checks.
+   * @see [[autoRefine]], [[refine]].
+   */
+  inline def unsafeAssume[B](using inline constraint: Constraint[A, B]): A :| B =
```

Comment:
I would remove the `unsafe` prefix (-> `assume[B]`) because I think the usage of such function is pretty common (like the example provided in #108 ).